app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt

```diff
@@ -63,13 +61,30 @@ fun DashboardScreen(
     val state by viewModel.state.observeAsState(DashboardState())
     val snackBar by viewModel.snackBar.observeAsState()
     val navController = LocalNavHostController.current
+    var dialogOpened by remember { mutableStateOf(false) }
+    var showDialog by remember { mutableStateOf(false) }
 
+    if (showDialog){
+        DisplayAlertDialog(
+            title = "Upload Trees Soon",
+            message ="You have over 2000 trees, please upload as soon as you can.",
+            dialogOpened = !dialogOpened ,
+            onOkayClicked = {
+                dialogOpened = false
+                navController.navigate(NavRoute.UserSelect.route)
+            }
+        )
+    }
     Dashboard(
         state = state,
         snackBar = snackBar,
         onSyncClicked = { viewModel.sync() },
         onOrgClicked = { navController.navigate(NavRoute.Org.route) },
-        onCaptureClicked = { navController.navigate(NavRoute.UserSelect.route) },
+        onCaptureClicked = {
+            if (state.totalTreesToSync >=2000){
```

Comment:
We can move this check into a boolean method on the state object that returns true or false